### PR TITLE
Fix Windows build in tensor merge PR

### PR DESCRIPTION
**Pull request by ezyang**
Prior to this patch, the Windows build failed saying that it ran out of heap. This fixes the problem.

This is done by splitting up THTensor.cpp into more compilation units; furthermore, THTensorMath.cpp is further split into THTensorMath, THTensorMoreMath, and THTensorEvenMoreMath. Because I'm pressed for time I didn't try to split them based on theme; it's just purely positional based on the original file.

### CODE REVIEW

File: aten/src/TH/generic/THTensor.h [hunk 1]
```diff
@@ -103,6 +103,7 @@ TH_API void THTensor_(setStorage4d)(THTensor *self, THStorage *storage_, ptrdiff
 TH_API void THTensor_(narrow)(THTensor *self, THTensor *src, int dimension_, int64_t firstIndex_, int64_t size_);
 TH_API void THTensor_(select)(THTensor *self, THTensor *src, int dimension_, int64_t sliceIndex_);
 TH_API void THTensor_(transpose)(THTensor *self, THTensor *src, int dimension1_, int dimension2_);
+TH_API int THTensor_(isTransposed)(const THTensor *self);
```
Comment (SsnL) [inline aten/src/TH/generic/THTensor.h]: why is this added?

**Review by gchanan — APPROVED** [review]
If this unblocks us I think it can go in.  But these file names are pretty bad.  THC has better names and you can just copy them.

File: aten/src/TH/generic/THTensor.h (hunk 1 above)
Comment (ezyang) [reply aten/src/TH/generic/THTensor.h]: It was used in MoreMath across the boundary, and I didn't think hard about how to get this function across the boundary. So into the header it went.

**Review by ssnl — APPROVED** [review]
lgtm to me as well. let's get this in to unblock.
